Controllers/TodoController.cs

```diff
@@ -9,40 +9,62 @@ public static class TodoController
             ITodoRepository todoRepository
         )
         {
-            throw new NotImplementedException();
+            var allTodos = await todoRepository.WhereAsync(todo => true);
+
+            return Results.Ok(
+                allTodos
+            );
         }
 
         public static async Task<IResult> GetTodo(
-           int id,
+           Guid id,
            ITodoRepository todoRepository
         )
         {
-            throw new NotImplementedException();
+            var todo = await todoRepository.FetchAsync(id);
+
+            if (todo == null)
+            {
+                return Results.NotFound();
+            }
+
+            return Results.Ok(todo);
         }
 
         public static async Task<IResult> CreateTodo(
             Todo todo,
             ITodoRepository todoRepository
         )
         {
-            throw new NotImplementedException();
+            var createdTodo = await todoRepository.CreateAsync(
+                todo
+            );
+
+            return Results.Ok(createdTodo);
         }
 
         public static async Task<IResult> UpdateTodo(
-            int id,
+            Guid id,
             Todo todo,
             ITodoRepository todoRepository
         )
         {
-            throw new NotImplementedException();
+            var updatedTodo = await todoRepository.UpdateAsync(
+                id,
+                todo
+            );
```

Comment:
That's it? Really?

Reply:
yes

Reply:
dont become a .NET fan ;)

Reply:
I don't think this is the fault of the languages or frameworks I'm used to. I think it's because of the projects I've worked on, where even seemingly simple requests like this require a `JOIN` or a loop somewhere.

Reply:
Not really sure what you mean :P